<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE bugzilla SYSTEM "https://bugzilla.redhat.com/page.cgi?id=bugzilla.dtd">

<bugzilla version="4.4.12068.1"
          urlbase="https://bugzilla.redhat.com/"
          
          maintainer="bugzilla-error-list@redhat.com"
>

    <bug>
          <bug_id>1309307</bug_id>
          
          <creation_ts>2016-02-17 07:37:00 -0500</creation_ts>
          <short_desc>python-trollius-redis: Provide a Python 3 subpackage</short_desc>
          <delta_ts>2016-03-03 14:12:21 -0500</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>2</classification_id>
          <classification>Fedora</classification>
          <product>Fedora</product>
          <component>python-trollius-redis</component>
          <version>24</version>
          <rep_platform>Unspecified</rep_platform>
          <op_sys>Unspecified</op_sys>
          <bug_status>CLOSED</bug_status>
          <resolution>RAWHIDE</resolution>
          
          
          <bug_file_loc></bug_file_loc>
          <status_whiteboard></status_whiteboard>
          <keywords></keywords>
          <priority>unspecified</priority>
          <bug_severity>unspecified</bug_severity>
          <target_milestone>---</target_milestone>
          
          <blocked>1285816</blocked>
          <everconfirmed>1</everconfirmed>
          <reporter name="Petr Viktorin">pviktori</reporter>
          <assigned_to name="Pierre-YvesChibon">pingou</assigned_to>
          <cc>pingou</cc>
          <qa_contact name="Fedora Extras Quality Assurance">extras-qa</qa_contact>
          
          <cf_fixed_in></cf_fixed_in>
          <cf_doc_type>Bug Fix</cf_doc_type>
          <cf_release_notes></cf_release_notes>
          <cf_story_points>---</cf_story_points>
          
          <cf_environment></cf_environment>
          <cf_last_closed>2016-03-03 14:12:21</cf_last_closed>
          <cf_type>Bug</cf_type>
          <cf_regression_status>---</cf_regression_status>
          <cf_mount_type>---</cf_mount_type>
          <cf_documentation_action>---</cf_documentation_action>
          <cf_crm></cf_crm>
          <cf_verified_branch></cf_verified_branch>
          <cf_category>---</cf_category>
          <cf_ovirt_team>---</cf_ovirt_team>
          
          <cf_cloudforms_team>---</cf_cloudforms_team>
          
          
          
          
          <target_release>---</target_release>
          
          <votes>0</votes>

      

      

      

          <comment_sort_order>oldest_to_newest</comment_sort_order>  
          <long_desc isprivate="0" >
    <commentid>9071808</commentid>
    <comment_count>0</comment_count>
    <who name="Petr Viktorin">pviktori</who>
    <bug_when>2016-02-17 07:37:31 -0500</bug_when>
    <thetext>Upstream, this software supports Python 3. Please provide a Python 3
package for Fedora.


According to the Python packaging guidelines [0], software must be
packaged for Python 3 if upstream supports it.
The guidelines give detailed information on how to do this, and even
provide an example spec file [1].

The current best practice is to provide subpackages for the two Python
versions (called &quot;Common SRPM&quot; in the guidelines). Alternatively, if
nothing depends on your Python2 package, you can just switch to Python 3
entirely.

It&apos;s fine to do this in Rawhide only.


If anything is unclear, or if you need any kind of assistance with the
porting, you can ask on IRC (#fedora-python on Freenode), or reply here.
We&apos;ll be happy to help!


[0] https://fedoraproject.org/wiki/Packaging:Python
[1] https://fedoraproject.org/wiki/Packaging:Python#Example_common_spec_file</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9072102</commentid>
    <comment_count>1</comment_count>
    <who name="Pierre-YvesChibon">pingou</who>
    <bug_when>2016-02-17 09:01:00 -0500</bug_when>
    <thetext>I am not sure this is going to happen, redis is a py2 port of asyncio, I really do not see the point in having a py3 version of a port of a py3 library.

So I&apos;ll wait a little before closing, but my first impression is WONTFIX :)</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9072156</commentid>
    <comment_count>2</comment_count>
    <who name="Pierre-YvesChibon">pingou</who>
    <bug_when>2016-02-17 09:20:41 -0500</bug_when>
    <thetext>Err, sorry not redis, trollius

And there is a asyncio-redis library: https://github.com/jonathanslenders/asyncio-redis which would be py3 only.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9072188</commentid>
    <comment_count>3</comment_count>
    <who name="Petr Viktorin">pviktori</who>
    <bug_when>2016-02-17 09:30:08 -0500</bug_when>
    <thetext>While Trollius is a port of asyncio, it&apos;s not Python2-only. It allows writing asyncio-style packages that are source-compatible with both py2 and py3, but anything based on it can&apos;t simply switch to asyncio when it drops py2 compatibility.
On the other hand, nothing in Fedora depends on trollius-redis, it probably isn&apos;t widely adopted elsewhere, and Trollius itself is deprecated* upstream.
* (we&apos;d say &quot;orphaned&quot; in Fedora)</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9072249</commentid>
    <comment_count>4</comment_count>
    <who name="Pierre-YvesChibon">pingou</who>
    <bug_when>2016-02-17 09:47:21 -0500</bug_when>
    <thetext>&gt; nothing in Fedora depends on trollius-redis

Pagure does :)

Trollius is orphaned/deprecated? I thought its upstream now works at RH?

That being said, you do have a point about the py2/py3 compatibility, I&apos;ll see about updating the package then.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9072413</commentid>
    <comment_count>5</comment_count>
    <who name="Petr Viktorin">pviktori</who>
    <bug_when>2016-02-17 10:35:09 -0500</bug_when>
    <thetext>http://trollius.readthedocs.org/deprecated.html

Victor does work at RH, but it was his personal project.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9094144</commentid>
    <comment_count>6</comment_count>
    <who name="Jan Kurik">jkurik</who>
    <bug_when>2016-02-24 10:24:31 -0500</bug_when>
    <thetext>This bug appears to have been reported against &apos;rawhide&apos; during the Fedora 24 development cycle.
Changing version to &apos;24&apos;.

More information and reason for this action is here:
https://fedoraproject.org/wiki/Fedora_Program_Management/HouseKeeping/Fedora24#Rawhide_Rebase</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9119081</commentid>
    <comment_count>7</comment_count>
    <who name="Pierre-YvesChibon">pingou</who>
    <bug_when>2016-03-03 14:12:21 -0500</bug_when>
    <thetext>Fixed in rawhide: http://koji.fedoraproject.org/koji/taskinfo?taskID=13216745

I will likely update F24 an F23 as well since the main change is the py3 package.</thetext>
  </long_desc>
      
      

    </bug>

</bugzilla>